docs/example-manifests/SLEMicro/dhcp/rke2-agent.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: sample-cluster
+  name: sample-cluster
+  namespace: default
+spec:
+  clusterName: sample-cluster
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: sample-cluster
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: sample-cluster
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha1
+          kind: RKE2ConfigTemplate
+          name: sample-cluster-workers
+      clusterName: sample-cluster
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: Metal3MachineTemplate
+        name: sample-cluster-workers
+      nodeDrainTimeout: 0s
+      version: v1.25.3+rke2r1
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha1
+kind: RKE2ConfigTemplate
+metadata:
+  name: sample-cluster-workers
+  namespace: default
+spec:
+  template:
+    spec:
+      agentConfig:
+        version: v1.25.3+rke2r1
+        kubelet:
+          extraArgs:
+            - provider-id=metal3://{{ ds.meta_data.uuid }}
+        nodeName: '{{ ds.meta_data.local_hostname }}'
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3MachineTemplate
+metadata:
+  name: sample-cluster-workers
+  namespace: default
+spec:
+  template:
+    spec:
+      dataTemplate:
+        name: sample-cluster-workers-template
+      hostSelector:
+        matchLabels:
+          cluster-role: worker
+      image:
+        checksum: http://imagecache.local:8080/SLE-Micro-eib-output.raw.md5sum
+        checksumType: md5sum
+        format: raw
+        url: http://imagecache.local:8080/SLE-Micro-eib-output.raw
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3DataTemplate
+metadata:
+  name: sample-cluster-workers-template
+  namespace: default
+spec:
+  clusterName: sample-cluster
+  metaData:
+    objectNames:
+      - key: name
+        object: machine
+      - key: local-hostname
+        object: machine
+      - key: local_hostname
+        object: machine
+  networkData:
+    links:
+      ethernets:
+        - id: eth0
+          macAddress:
+            fromHostInterface: eth0
+          type: phy
+    networks:
+      ipv4DHCP:
+        - id: eth0
+          link: eth0
+          routes:
```

Comment:
I think this is from the original example for the capi yaml. The parameter is optional:
https://github.com/metal3-io/cluster-api-provider-metal3/blob/1dd837ee1fa624aea875d953392d94b932412b09/api/v1alpha5/metal3datatemplate_types.go#L380C3-L380C3
IMHO, if we remove it because it's dhcp and I agree with you, we should remove also from OpenSuseLeap.
I just kept the same to make it consistent, but I will do a test removing it, and I will change if the tests are succesfully.

Reply:
It was needed for Leap. No clue if it's necessary for SLE Micro.

Reply:
did you try without that part and it didn't work fine on Leap?

Reply:
Yes, I did. The CAPI components required a valid value for the gateway and the `network: 0.0.0.0` in order to bring up the network interfaces properly.

Reply:
This was before @hardys's huge refactoring, though. Not sure how that affected the infrastructure components and if we can get rid of this.

Reply:
I wouldn't recommend modifying the Leap manifests here. It's quite a lot of work to test, debug and verify the workflow so it should be done in a separate PR.

I don't mind if you want to experiment with the SLE Micro manifests here, though. Your call.

Reply:
ok thanks for the info Atanas, I will try on the SLE Micro just to know if we can change it or not

Reply:
tested by Edu and me also...working fine without that part, so I've removed it